src/main/java/io/github/bookcrawler/EmpikBookExtractor.java

```diff
@@ -0,0 +1,60 @@
+package io.github.bookcrawler;
+
+import io.github.bookcrawler.entities.BookInfo;
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class EmpikBookExtractor implements BookExtractor {
+
+    public Iterable<BookInfo> extract() throws IOException {
+        Connection connect = Jsoup.connect("http://www.empik.com/ebooki/promocje/");
+        Set<BookInfo> books = new HashSet<BookInfo>();
+
+
+        Document document = connect.get();
+        Elements contentPacket = document.getElementsByAttributeValue("class", "more_lnk");
+
+        List<String> sectionUrls = new ArrayList<String>();
+        for (Element element : contentPacket) {
+            sectionUrls.add("http://www.empik.com" + element.attr("href"));
+        }
+
+        for (String url : sectionUrls) {
+            books.addAll(getBooksFromSection(connect.url(url).get()));
+        }
```

Comment:
As mentioned before. It will be refactored. I will use your hints into stream. Thank you.

Nevertheless I don't see O(n^2) in this. Its one loop after another. Not nested. I think O(2n). Still It's gonna change.